lib/InitModelTypes/InitModelTypes.cpp

```diff
@@ -95,13 +97,23 @@ static RecursiveCoroutine<bool> addOperandType(const llvm::Value *Operand,
         if (OperandType->isPointer()) {
           // If the operand is already a pointer, just forward it
           TypeMap.insert({ Operand, OperandType.copy() });
-
         } else if (not PointersOnly) {
           auto PS = model::Architecture::getPointerSize(Model.Architecture());
           TypeMap.insert({ Operand, model::PrimitiveType::makeGeneric(PS) });
+        } else {
+          // WIP: ??
         }
         rc_return true;
       }
+    } else if (Expr->getOpcode() == Instruction::PtrToInt) {
+      auto It = TypeMap.find(Expr->getOperand(0));
+      if (It != TypeMap.end()) {
+        const model::UpcastableType &OperandType = It->second;
+        // WIP: PointersOnly?
```

Comment:
@pfez This is not a pointer, shall I insert it only `if (not PointersOnly)`?

Reply:
Forget about all this handling. The handling of `PtrToInt` should be exactly the same as `IntToPtr` AFAICT.

Basically if the operand type is a pointer, you add it to the map regardless.

Otherwise, you `makeGeneric` and add it to the map, but only when `not PointersOnly`, because if `PointersOnly` is true you would be adding a non-pointer to the map which is wrong.